src/userscripts.ts

```diff
@@ -109,14 +138,15 @@ class Userscript implements IUserscriptInstance {
 
 }
 
-ipcRenderer.on('main_initializes_userscripts', (event, recieved_userscriptsPath: string) => {
-	su.userscriptsPath = recieved_userscriptsPath;
+ipcRenderer.on('main_initializes_userscripts', (event, recieved_userscript_paths: { userscriptsPath: string, userscriptPrefsPath: string }) => {
+	su.userscriptsPath = recieved_userscript_paths.userscriptsPath;
 	su.userscriptTrackerPath = pathResolve(su.userscriptsPath, 'tracker.json');
+	su.userscriptPrefsPath = recieved_userscript_paths.userscriptPrefsPath
 
 	// init the userscripts (read, map and set up tracker)
 	su.userscripts = readdirSync(su.userscriptsPath, { withFileTypes: true })
 		.filter(entry => entry.name.endsWith('.js'))
-		.map(entry => new Userscript({ name: entry.name, fullpath: pathResolve(su.userscriptsPath, entry.name).toString() }));
+		.map(entry => new Userscript({ name: entry.name, settingsPath: pathResolve(su.userscriptPrefsPath, entry.name.replace(/.js$/, '.json')), fullpath: pathResolve(su.userscriptsPath, entry.name).toString() }));
```

Comment:
i am not sure this is a good idea; not every valid js object is valid json? i'd argue most aren't. 
What is the reasoning behind this?

Reply:
This code is replacing the .js in fileName.js with .json, so that the client will automatically generate a unique .json file whenever a custom option is changed. The validation for this was done in code before this, ensuring it's a .js file by scanning the directory for them, and uniqueness because no two files can have the exact same name, this is the reason the validation runs twice, once on startup and once on setting application, just in case someone changed their script's filename.

I should probably leave comments around...

Reply:
ah, i misunderstaood what this does. it does look a tiny bit confusing, but the logic is fine.